chunky/src/java/se/llbit/chunky/ui/controller/WorldChooserController.java

```diff
@@ -186,17 +188,25 @@ private void fillWorldList(final File worldSavesDir) {
     statusLabel.setText("Loading worlds list...");
     disableControls(true);
 
-    Task<List<World>> loadWorldsTask = new Task<List<World>>() {
+    Task<List<World>> loadWorldsTask = new Task<>() {
       @Override
       protected List<World> call() {
         List<World> worlds = new ArrayList<>();
         if (worldSavesDir != null) {
           File[] worldDirs = worldSavesDir.listFiles();
           if (worldDirs != null) {
             for (File dir : worldDirs) {
-              if (World.isWorldDir(dir)) {
-                worlds.add(World.loadWorld(dir, World.OVERWORLD_DIMENSION,
-                    World.LoggedWarnings.SILENT));
+              for (WorldFormat worldFormat : WorldFormat.worldFormats) {
+                if (worldFormat.isValid(dir.toPath())) {
+                  try {
+                    World world = worldFormat.loadWorld(dir.toPath(), String.valueOf(World.OVERWORLD_DIMENSION));
+                    if (world != EmptyWorld.INSTANCE) {
+                      worlds.add(world);
+                    }
+                  } catch (IOException e) {
+                    throw new RuntimeException(e);
```

Comment:
Log this instead? Listing worlds shouldn't crash Chunky just because there's a bad world.